File: src/main/java/com/api/prisma_vi/utils/apiError/HttpErrorHandler.java
package com.api.prisma_vi.utils.apiError;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.MethodArgumentNotValidException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import java.util.List;
import java.util.stream.Collectors;

@RestControllerAdvice
public class HttpErrorHandler {


    @ExceptionHandler(MethodArgumentNotValidException.class)
    public ResponseEntity<ErrorView<List<ErrorFieldView>>> treatCode400(MethodArgumentNotValidException ex) {
        List<ErrorFieldView> errorFields = ex.getBindingResult().getFieldErrors().stream()
                .map(error -> new ErrorFieldView(error.getField(), error.getDefaultMessage()))
                .collect(Collectors.toList());

        ErrorView<List<ErrorFieldView>> errorView = new ErrorView<>(HttpStatus.NOT_ACCEPTABLE, errorFields);
        return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(errorView);
    }

}

File: src/main/java/com/api/prisma_vi/apiError/HttpErrorHandler.kt
package com.api.prisma_vi.apiError

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice


@RestControllerAdvice
class HttpErrorHandler {

    @ExceptionHandler(MethodArgumentNotValidException::class)
    fun treatCode400(ex: MethodArgumentNotValidException): ResponseEntity<ErrorView<List<ErrorFieldView>>> =
        ResponseEntity.status(406).body(
            ErrorView(
                HttpStatus.NOT_ACCEPTABLE,
                ex.fieldErrors.map { ErrorFieldView(it.field, it.defaultMessage) })
        )

}